arrows/vxl/average_frames.cxx

```diff
@@ -0,0 +1,731 @@
+    }
+
+    if( type == "window")
+    {
+      if( is_byte )
+      {
+        byte_averager.reset(
+          new windowed_frame_averager< vxl_byte >( round, window_size ) );
+      }
+      else
+      {
+        float_averager.reset( 
+          new windowed_frame_averager< double >( round, window_size ) );
+      }
+    }
+    else if( type == "cumulative" )
+    {
+      if( is_byte )
+      {
+        byte_averager.reset(
+          new cumulative_frame_averager< vxl_byte >( round ) );
+      }
+      else
+      {
+        float_averager.reset(
+          new cumulative_frame_averager< double >( round ) );  
+      }
+    }
+    else if( type == "exponential" )
+    {
+      if( exp_weight <= 0 || exp_weight >= 1 )
+      {
+        throw std::runtime_error( "Invalid exponential averaging coefficient!" );
+      }
+
+      if( is_byte )
+      {
+        byte_averager.reset(
+          new exponential_frame_averager< vxl_byte >( round, exp_weight ) );
+      }
+      else
+      {
+        float_averager.reset(
+          new exponential_frame_averager< double >( round, exp_weight ) );   
+      }
+    }
+    else
+    {
+      throw std::runtime_error( "Invalid averaging type!" );
+    }
+  }
+
+  // Compute the updated average with the current frame
+  // return the average or the variance
+  kwiver::vital::image_container_sptr
+  process_frame( vil_image_view< double > input )
+  {
+    load_model( false );
+
+    if( !output_variance )
+    {
+      vil_image_view< double > output;
+      float_averager->process_frame( input, output );
+      return std::make_shared< vxl::image_container >( output );
+    }
+    else
+    {
+      vil_image_view< double > tmp, output;
+      float_averager->process_frame( input, tmp, output );
+      return std::make_shared< vxl::image_container >( output );
+    }
+  }
+};
+
+// ----------------------------------------------------------------------------
+average_frames
+::average_frames()
+: d( new priv() )
+{
+  attach_logger( "arrows.vxl.average_frames" );
+}
+
+// ----------------------------------------------------------------------------
+average_frames
+::~average_frames()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+average_frames
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "type", d->type,
+    "Operating mode of this filter, possible values: cumulative, window, "
+    "or exponential." );
+  config->set_value( "window_size", d->window_size,
+    "The window size if computing a windowed moving average." );
+  config->set_value( "exp_weight", d->exp_weight,
+    "Exponential averaging coefficient if computing an exp average." );
+  config->set_value( "round", d->round,
+    "Should we spend a little extra time rounding when possible?" );
+  config->set_value( "output_variance", d->output_variance,
```

Comment:
This seems to be in conflict with the actual behavior, which seems to always output variance as a double. However, as @mleotta mentioned, the `average` image can be a byte, which seems odd.

Reply:
I believe that variance is a configurable feature in the [BurnOut feature extractor super process](https://github.com/Kitware/burn-out/blob/1d2afcd36d37127b5a3b6fa3f60bc195408ab3ea/library/object_detectors/pixel_feature_extractor_super_process.txx). In the default config I've seen, it is on. So my understanding is it is being used.